cmd/openshift-install/internal_integration_test.go

```diff
@@ -103,11 +104,68 @@ func runIntegrationTest(t *testing.T, testFolder string) {
 				}
 			}
 
-			// Let's get the current release version, so that
-			// it could be used within the tests
-			pullspec, err := releaseimage.Default()
-			if err != nil {
-				return err
+			var pullspec string
+			// If set, let's use $OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE to replace test data
+			// and also pass it to the testscript environment. It will be usually set in a CI job
+			// to reference the ephemeral payload release.
+			if releaseImageOverride, ok := os.LookupEnv("OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE"); ok && releaseImageOverride != "" {
+				pullspec = releaseImageOverride
+				e.Vars = append(e.Vars, fmt.Sprintf("OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE=%s", pullspec))
+				if authFilePath, ok := os.LookupEnv("AUTH_FILE"); ok && authFilePath != "" {
+					// Read the new pull secret from auth.json
+					authFile, err := os.ReadFile(authFilePath)
+					if err != nil {
+						return err
+					}
+					var auths map[string]interface{}
+					if err := json.Unmarshal(authFile, &auths); err != nil {
+						t.Logf("Error unmarshalling auth.json: %v\n", err)
+						return err
+					}
+					newPullSecret, err := json.Marshal(auths)
+					if err != nil {
+						return err
+					}
+
+					// Read the existing install-config.yaml
+					workDir := e.Getenv("WORK")
+					installConfigPathAbs := filepath.Join(workDir, "install-config.yaml")
+					installConfigFile, err := os.ReadFile(installConfigPathAbs)
+					if err != nil {
+						if os.IsNotExist(err) {
+							// skip the error when test does not required reading the install-config.yaml
+							return nil
+						}
+						// For other errors, return them
+						return err
+					}
+					var config map[string]interface{}
+					if err := yaml.Unmarshal(installConfigFile, &config); err != nil {
+						t.Logf("Error unmarshalling install-config.yaml: %v\n", err)
+						return err
+					}
+
+					// Update the pullSecret
+					config["pullSecret"] = string(newPullSecret)
+
+					// Write the updated config back to install-config.yaml
+					updatedConfig, err := yaml.Marshal(&config)
+					if err != nil {
+						return err
+					}
+					if err := os.WriteFile(installConfigPathAbs, updatedConfig, 0600); err != nil {
+						t.Logf("Error writing updated install-config.yaml: %v\n", err)
+						return err
+					}
+					t.Log("PullSecret replaced successfully.")
+				}
+			} else {
+				// Let's get the current release version, so that
+				// it could be used within the tests
+				pullspec, err = releaseimage.Default()
```

Comment:
Wouldn't doing

```
if releaseImageOverride, ok := os.LookupEnv("OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE"); ok && releaseImageOverride != "" {
        pullspec = releaseImageOverride
} else {
        if pullspec, err = releaseimage.Default(); err != nil {
                return err
        }
}
```

eliminate the need for `replaceReleaseImage()`?

Reply:
no, I had to replace it.

Reply:
> no, I had to replace it.

testscripts applies env var substitution for the script file lines, but _not_ for the generated files. For that it usually offers cmd with an alternative flavor - where the env vars are substituted (named with an `env` suffix, such as [cmpenv](https://pkg.go.dev/github.com/rogpeppe/go-internal/testscript)).

That's the reason why I've added [env var expansion](https://github.com/openshift/installer/blob/5ddaa22b836479367bb00b9fb3c0d43e924b6ee8/cmd/openshift-install/internal_integration_test.go#L311) in our commands, so my expectation is that every (internal) testscript command that is based on `byteCompare`, such as `isoCmp`, will already support a generic mechanism for env var expansion. So, as long the `RELEASE_IMAGE` is properly filled with a value, `replaceReleaseImage` should not be required